erpnext/patches/v13_0/updates_for_multi_currency_payroll.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2019, Frappe and Contributors
+# License: GNU General Public License v3. See license.txt
+
+import frappe, erpnext
+from frappe.model.utils.rename_field import rename_field
+
+def execute():
+
+	frappe.reload_doc('Accounts', 'doctype', 'Salary Component Account')
+	if frappe.db.has_column('Salary Component Account', 'default_account'):
+		rename_field("Salary Component Account", "default_account", "account")
+
+	doctype_list = [
+		{
+		'module':'HR',
+		'doctype':'Employee Advance'
+		},
+		{
+		'module':'HR',
+		'doctype':'Leave Encashment'
+		},
+		{
+		'module':'Payroll',
+		'doctype':'Additional Salary'
+		},
+		{
+		'module':'Payroll',
+		'doctype':'Employee Benefit Application'
+		},
+		{
+		'module':'Payroll',
+		'doctype':'Employee Benefit Claim'
+		},
+		{
+		'module':'Payroll',
+		'doctype':'Employee Incentive'
+		},
+		{
+		'module':'Payroll',
+		'doctype':'Employee Tax Exemption Declaration'
+		},
+		{
+		'module':'Payroll',
+		'doctype':'Employee Tax Exemption Proof Submission'
+		},
+		{
+		'module':'Payroll',
+		'doctype':'Income Tax Slab'
+		},
+		{
+		'module':'Payroll',
+		'doctype':'Payroll Entry'
+		},
+		{
+		'module':'Payroll',
+		'doctype':'Retention Bonus'
+		},
+		{
+		'module':'Payroll',
+		'doctype':'Salary Structure'
+		},
+		{
+		'module':'Payroll',
+		'doctype':'Salary Structure Assignment'
+		},
+		{
+		'module':'Payroll',
+		'doctype':'Salary Slip'
+		},
+	]
+
+	base_keys = [
+		{
+			"base_key": "base_hour_rate",
+			"key": "hour_rate"
+		},
+		{
+			"base_key": "base_gross_pay",
+			"key": "gross_pay"
+		},
+		{
+			"base_key": "base_total_deduction",
+			"key": "total_deduction"
+		},
+		{
+			"base_key": "base_net_pay",
+			"key": "net_pay"
+		},
+		{
+			"base_key": "base_rounded_total",
+			"key": "rounded_total"
+		},
+		{
+			"base_key": "base_total_in_words",
+			"key": "total_in_words"
+		},
+	]
+
+	for item in doctype_list:
+		frappe.reload_doc(item['module'], 'doctype', item['doctype'])
+		all_doc = frappe.get_all(item['doctype'])
+		if all_doc:
+			for record in all_doc:
```

Comment:
we need to avoid for loop and use a single SQL query to update currency, otherwise it can take hours to run the patch